src/utils/price.ts

```diff
@@ -0,0 +1,77 @@
+import BN from 'bn.js'
+import { assert } from './miscellaneous'
+import { TEN, UNLIMITED_ORDER_AMOUNT } from 'const'
+import BigNumber from 'bignumber.js'
+
+export function adjustAmountToLowerPrecision(amount: BN, higherPrecision: number, lowerPrecision: number): BN {
+  if (higherPrecision === lowerPrecision) {
+    // no adjustment required
+    return amount
+  }
+
+  assert(higherPrecision > lowerPrecision, 'higherPrecision must be > lowerPrecision')
+
+  const difference = new BN(higherPrecision - lowerPrecision)
+
+  // divides amount by difference in precision, rounding. E.g:
+  // 1.2345, precision 4, amount == 12345
+  // 1.2345, precision 2, amount == 12345 / 10 ^ (4 - 2) => 12345 / 100 => 123
+  // 1.5550, precision 1, amount == 15550 / 10 ^ (4 - 1) => 15550 / 1000 => 16
+  return amount.divRound(TEN.pow(difference))
+}
+
+function bigNumberToBN(n: BigNumber | BN): BN {
+  if (n instanceof BN) {
+    return n
+  }
+  return new BN(n.integerValue().toString(10))
+}
+
+interface Amounts {
+  buyAmount: BN
+  sellAmount: BN
+}
+
+const MAX = new BigNumber(UNLIMITED_ORDER_AMOUNT.toString())
```

Comment:
inside `/const` :smile:

Reply:
Well, that amount is already there as BN. I 'duplicated' it here for local usage only.
But you are right, will move it there.